internal/handshake/cipher_suite.go

```diff
@@ -71,15 +88,46 @@ func aeadChaCha20Poly1305(key, nonceMask []byte) *xorNonceAEAD {
 // xorNonceAEAD wraps an AEAD by XORing in a fixed pattern to the nonce
 // before each call.
 type xorNonceAEAD struct {
-	nonceMask [aeadNonceLength]byte
-	aead      cipher.AEAD
+	nonceMask        [aeadNonceLength]byte
+	aead             cipher.AEAD
+	hasSeenNonceZero bool // This value denotes if the aead field was used with a nonce = 0
 }
 
 func (f *xorNonceAEAD) NonceSize() int        { return 8 } // 64-bit sequence number
 func (f *xorNonceAEAD) Overhead() int         { return f.aead.Overhead() }
 func (f *xorNonceAEAD) explicitNonceLen() int { return 0 }
 
+func allZeros(nonce []byte) bool {
+	for _, e := range nonce {
+		if e != 0 {
+			return false
+		}
+	}
+	return true
+}
+
 func (f *xorNonceAEAD) Seal(out, nonce, plaintext, additionalData []byte) []byte {
+	if useBoring && boring.Enabled() && !f.hasSeenNonceZero {
```

Comment:
Should we have a similar check like https://github.com/chungthuang/quic-go/pull/4/files#diff-98397ba38acdcfbbad83baae17287dceb2e0fad20ec50ad4f049943db6a5d5ddR60?

Reply:
I will make it similar and in case we have useBoring = true and boring.Enabled() = false the program will panic

Reply:
done